### ISSUE TITLE
Invalid error messages for scalar arrays

### ISSUE THREAD

**Issue by hapcode** [async-graphql/async-graphql#338]
## Expected Behavior

Output 1 for `fewReq: [DateTime!]!` (should have trailing exclamation):
```json
{
    "message": "Failed to parse \"DateTime\": trailing input (occurred while parsing \"[DateTime!]!\") (occurred while parsing \"TestInput\")",
}
```

Output 2 for `fewOpt: [DateTime!]` (should have no duplicates):
```json
{
    "message": "Failed to parse \"DateTime\": trailing input (occurred while parsing \"[DateTime!]\") (occurred while parsing \"TestInput\")",
}
```

## Actual Behavior

Output 1 for `fewReq: [DateTime!]!` (missing trailing exclamation):
```json
{
    "message": "Failed to parse \"DateTime\": trailing input (occurred while parsing \"[DateTime!]\") (occurred while parsing \"TestInput\")",
}
```

Output 2 for `fewOpt: [DateTime!]` (duplicates):
```json
{
    "message": "Failed to parse \"DateTime\": trailing input (occurred while parsing \"[DateTime!]\") (occurred while parsing \"[DateTime!]\") (occurred while parsing \"TestInput\")",
}
```

## Steps to Reproduce the Problem

SDL and code:
```gql
input TestInput {
    fewReq: [DateTime!]!
    fewOpt: [DateTime!]
}
```
```rust
#[derive(InputObject)]
pub struct TestInput {
    few_req: Vec<DateTime<Utc>>,
    few_opt: Option<Vec<DateTime<Utc>>>,
}
```

Query:
```gql
mutation Test($input: TestInput!) {
  test(input: $input)
}
```

Input 1:
```json
{ 
  "input": { 
    "fewReq": ["2020-01-01T00:00:00+00:00err"],
    "fewOpt": ["2020-01-01T00:00:00+00:00"]
  } 
}
```

Input 2:
```json
{ 
  "input": { 
    "fewReq": ["2020-01-01T00:00:00+00:00"],
    "fewOpt": ["2020-01-01T00:00:00+00:00err"]
  } 
}
```

## Specifications

- Version: 2.0.12
- Platform: MacOS

**Comment by sunli829** [comment]
@Koxiaet When you have a moment, would you please help me look at this problem? 🙂

**Comment by Koxiaet** [comment]
Should be fixed now with 9d94c93.

**Comment by hapcode** [comment]
> Should be fixed now with [9d94c93](https://github.com/async-graphql/async-graphql/commit/9d94c933ff442fe9c5545fafca134d6a199966e6).

I don't want to sound boring but this fix is for the Output 2. In `async-graphql 2.1.1` I have identical errors for both types `[DateTime!]!` (<-- trailing exclamation) and `[DateTime!]`. I don't think it's important, just my note.
